### DISCUSSION TITLE
How to use refresh token?

### DISCUSSION THREAD

**Discussion by TerrizX** [spotify-web-api-java/spotify-web-api-java#269] [Q&A] [answered]
I'm using the PKCE authorization method and I'm new to using access and refresh tokens. I followed the examples provided  and acquired the access and refresh token and I've used the access token to get stuff. But after the access token expires do you then do the last example (your Refresh Example) and use that refresh token as an access token? I'd appreciate some explanation of where to put your refresh token after your access token expires. Thanks in advance.

**Comment by dargmuesli** [comment, marked as answer]
Spotify's API documentation offers extensive explanations for that topic: https://developer.spotify.com/documentation/general/guides/authorization/code-flow/#request-a-refreshed-access-token

**Reply by TerrizX** [reply]
I understand but I'm talking more in the sense of where to place the refresh token. I've obtained a refresh token, but I just don't know where I put it to call endpoints again.

**Reply by dargmuesli** [reply]
Use it in the refresh request to get new tokens: https://github.com/spotify-web-api-java/spotify-web-api-java/blob/d3dfa254aac3082e1ad055c629ebf88ccfb414f2/examples/authorization/authorization_code/pkce/AuthorizationCodePKCERefreshExample.java
